### ISSUE TITLE
Ubuntu Linux Storage management issues

### ISSUE THREAD

**Issue by EonDigital** [ValveSoftware/steam-for-linux#9797]
#### Your system information

* Steam client version (build number or date): 
Steam Version:  1688171965
Steam Client Build Date:  Fri, Jun 30 5:19 PM UTC -08:00
Steam Web Build Date:  Fri, Jun 30 4:51 PM UTC -08:00
Steam API Version:  SteamClient020
* Distribution (e.g. Ubuntu): 
Ubuntu
* Opted into Steam client beta?: [Yes/No]
Yes
* Have you checked for system updates?: [Yes/No]
Yes
* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`]
[steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/11952584/steam-logs.tar.gz)
* GPU: Nvidia/Intel/AMD
Nvidia

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

Attempting to add a new storage location runs into several problems.
I am unable to add locations outside the ~/.steam directory.  I also identified a crash.  These are two distinct issues, but they're closely related, so I will document both here.

My expectation was I should be able to add a volume that is elsewhere on my system.  I have one full drive and a second mounted that would be a preferred place to store games.

I also expected that interactions with this window would not result in a crash.  Fine if selecting nothing is invalid, but it should fail gracefully.

#### Steps for reproducing this issue:

1. Open settings/Storage
2. Click the + to add storage
3. The directory structure visible only includes the directories contained in ~/.steam
4. In addition to being unable to navigate out of .steam, symlinks to any directory outside .steam do not show in the list, preventing a simple workaround. (issue 1)
5. If you don't select a folder and click "Select", the steam client crashes. (issue 2)

**Comment by kisak-valve** [comment]
Hello @EonDigital, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

Previously mentioned at https://github.com/ValveSoftware/steam-for-linux/issues/9377#issuecomment-1615828136.

**Comment by AdamFreemen** [comment]
@EonDigital you can try the following:
1. Launch Steam with the -console option on the command line.
2. Click the Console tab.
3. Run the command "library_folder_add /path_to_your_folder/" without quotes.

The output should be: "Mounted path /path_to_your_folder/"

**Comment by EonDigital** [comment]
I'm unblocked at this point.  I repartitioned my hard drive before this comment.  I can at least confirm the console workaround does what is intended... I'm able to add the new location.

**Comment by segaretroboy** [comment]
@EonDigital tysvm, Steam lost my external library and this let me re-add it! Appreciate it a lot!

**Comment by EonDigital** [comment]
@AdamFreemen - I think that thank you goes your way.

@kisak-valve - I've already posted the diagnostics I'm comfortable posting for this issue.  I can confirm part of this issue is a duplicate of #9377, but it does not mention the crash when nothing is selected.

![Screenshot from 2023-07-02 15-35-14](https://github.com/ValveSoftware/steam-for-linux/assets/20853321/648e8e28-59b9-4aff-b100-55c81b2e1a20)
![Screenshot from 2023-07-02 15-44-52](https://github.com/ValveSoftware/steam-for-linux/assets/20853321/97f10a9c-4215-45c6-83dd-446b5ae92ec8)

**Comment by AdamFreemen** [comment]
@EonDigital better way is record video with OBS. You can install latest version from flatpak.

**Comment by EonDigital** [comment]
I trigger it by literally hitting the select button without selecting anything.  It's likely an issue in how the value the select dialog returns is consumed when it's expecting a non-null result.  If you don't see an issue in that code, just close this as not planned.

**Comment by sylware** [comment]
We cannot browse the parent directories: then I create a folder manually which will be added to the list, but when I select it, the client [beta] crashes.

(I removed one of my storage directory, then cannot add it anymore).

**Comment by jackmillward** [comment]
I am seeing this on Void Linux, workaround as suggested by @AdamFreemen worked great.

The error message in the terminal for this problem is:
```
Unsupported desktop portal version: 0, falling back to a different file picker
```

**Comment by Irets** [comment]
I'm seeing the same issue on Gentoo Linux.
@AdamFreemen's suggestion worked.

I'm not using Flatpak, snaps or anything like that.

**Comment by metafarion** [comment]
> I'm seeing the same issue on Gentoo Linux.
> @AdamFreemen's suggestion worked.

2x ditto
Must be a UI glitch

**Comment by gabubonico** [comment]
I'm currently having the same problem, but in my case I want to add a Non-Steam game to my library so I can run it with Proton. Anyone can suggest a workaround?

**Comment by tTomer77** [comment]
I can confirm, that I have the same issue with _Archlinux_; on a brand new fresh installed PC.
Cannot navigate parent higher than  **~/.steam/root/**

I use as desktop decorator xfce (xfwm4 version 4.18.0 (revision 7e7473c5b), with thunar for file navigations. The steam client seems to open something which looks like a gtk window. For me it looks like a build-in navigation windows from the steam client.

Steam-Version:  1689034492
Steam-Client: Build-Datum:  Di., 11. Juli 1:59 UTC -08:00
Steam: Webbuild-Datum:  Mo., 10. Juli 19:48 UTC -08:00
Steam-API-Version:  SteamClient020

![missing_Navigation](https://github.com/ValveSoftware/steam-for-linux/assets/140399298/da5dccf2-2f1c-4318-8edf-608393654837)

The solution with steam console works also fine for me.
1. open bash terminal and start steam-client with the command   **steam -console**   or   **steam steam://open/console**   (whatever works for you)
2. Goto Console-Tab and enter the command  **library_folder_add   /path_to_your_folder/** (by the way, many thanks @AdamFreemen, it also worked for me).
The library is then normal visible/listed in the steam-preferences.

![addpath](https://github.com/ValveSoftware/steam-for-linux/assets/140399298/c54628fd-180c-487b-bf85-fee6ac535ab8)

Would be nice when this could be fixed in the linux steam-client that the workaround is not more needed.

**Comment by tTomer77** [comment]
@Murkless, check with "ls -la", if the folder is yours. I'm sure that /mnt folders belongs to root:root and not USER:users. Refer also the command for user owners "chown" (may be different, I don't know which linux distribution you have)

**Comment by Murkless** [comment]
@tTomer77 Hi. I've already fixed the issue by editing fstab file. If this  method is not described here I could write about it a bit later

**Comment by tTomer77** [comment]
@Murkless Okay, then you have already a solution for yourself. The location /mnt is normally not used for steam libraries. This is a location for root. The fstab is for partitioning and mounting. I can imagine that you have configured some bind.

However, for me my solution has worked. Form my point you don't need it to describe it here. Please be also carefully with posting sensible data/information's. Have a nice day

**Comment by Murkless** [comment]
@tTomer77 your solution didn't work for me. Error was something like "error mounting that path"
About using /mnt/. I have to driver in my PC, and second one is mounted /mnt/DataGames. It is very convenient and popular mount point. I didn't want to change the mount point only for steam

**Comment by Murkless** [comment]
@tTomer77 your solution didn't work for me. Error was something like "error mounting that path"
About using /mnt/. I have to driver in my PC, and second one is mounted /mnt/DataGames. It is very convenient and popular mount point. I didn't want to change the mount point only for steam

**Comment by Murkless** [comment]
So, as I said, I will post my solution. Here it is. 
Firstly, I ran steam from terminal so its logger connects to stdout. When trying to add another folder from the second drive I get something like "Permission denied". I've tried changing them by using chown and this gave no results. So I changed my fstab file and now it looks like this 
`# Static information about the filesystems.
# See fstab(5) for details.

# <file system> <dir> <type> <options> <dump> <pass>
# /dev/nvme0n1p3
UUID=2d04f5d2-6a98-4d1a-be2b-b59dffdd3004   /                ext4             rw,relatime         0 1 

# /dev/nvme0n1p2
UUID=30ff625e-7d9d-4469-a9e2-13cac18eac15   /boot            ext4             rw,relatime         0 2 

UUID=9aadfd4e-80d8-4f34-b413-66db1cfd542f   /mnt/DataGames   ext4             rw, suid, dev, exec, auto, nouser, and async   0 0 `
Hope this helps you as well
